cni/network/network.go

```diff
@@ -60,13 +61,26 @@ const (
 	jsonFileExtension = ".json"
 )
 
+type  ExecutionModes string
+const (
+    Default ExecutionModes = "default"
+    Baremetal ExecutionModes = "baremetal"
+)
+
 // NetPlugin represents the CNI network plugin.
 type netPlugin struct {
 	*cni.Plugin
 	nm          network.NetworkManager
 	ipamInvoker IPAMInvoker
 	report      *telemetry.CNIReport
 	tb          *telemetry.TelemetryBuffer
+	nnsClient   NnsClient
```

Comment:
I have moved the nns grpclient out of plugin. It is at the root level now just like other client packages. The reason it is part of netPlugin struct is that because we are doing a dependency injection here via an interface. And it is passed into plugin struct from main.go. This way we can also pass a mock implementation for testing. For example - we can add a parameter in CNI config saying - "UseMockNns" and when it is true, we initialize and pass a mock client all the way to plugin. That is the intention here.